canonical-web-and-design/maas-ui
# refactor(ui): generic node filtering

## Done

- Update the search and filter code to work with any node type.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the machine list and check you can still search and filter.
- Visit a KVM details page and check that you can still search the VMs.

## Fixes

Fixes: canonical-web-and-design/app-squad#136.
